### PR TITLE
checkpoint right after sky background rendering

### PR DESCRIPTION
**Pull request by jchiang87**
This addresses https://github.com/LSSTDESC/imSim/issues/172

### CODE REVIEW

**Comment by jchiang87** [comment]
I found a logic error in the scope where the checkpoint writing is made; I just pushed the fix.

File: python/lsst/sims/GalSimInterface/galSimInterpreter.py [hunk 1]
```diff
@@ -644,15 +645,18 @@ def write_checkpoint(self, force=False):
...
                                rng=self._rng,
-                               drawn_objects=self.drawn_objects,
+                               drawn_objects=drawn_objects,
                                centroid_objects=self.centroid_list)
             with tempfile.NamedTemporaryFile(mode='wb', delete=False,
                                              dir='.') as tmp:
                 pickle.dump(image_state, tmp)
                 tmp.flush()
                 os.fsync(tmp.fileno())
+                os.chmod(tmp.name, 0o660)
```
Comment (cwwalter) [inline python/lsst/sims/GalSimInterface/galSimInterpreter.py]: Just checking you wanted this to show up in this PR and it isn't left over from somewhere else.

Comment (cwwalter) [reply python/lsst/sims/GalSimInterface/galSimInterpreter.py]: Oops never mind.  I see this was in your PR description.

**Comment by danielsf** [comment]
Jenkins passed

**Pull request merged**